ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java

```diff
@@ -450,19 +470,9 @@ private void notifyForkChoiceUpdatedAndOptimisticSyncingChanged() {
             .getForkChoiceState(
                 recentChainData.getJustifiedCheckpoint().orElseThrow(),
                 recentChainData.getFinalizedCheckpoint().orElseThrow());
-    UInt64 latestFinalizedBlockSlot = recentChainData.getStore().getLatestFinalizedBlockSlot();
```

Comment:
@ajsutton I'm a bit uncomfortable about moving this later on notification reception. Can't currently figure out if we need to ensure consistency with `ForkChoiceState`.

Reply:
to be supersafe I could make `ForkChoiceUpdatedResultNotification` to be a context created here, in advance, that brings with if `ForkChoiceState` and `latestFinalizedBlockSlot`